### DISCUSSION TITLE
We use workers with Fargate, but there's a problem

### DISCUSSION THREAD

**Discussion by 5pponent** [apache/airflow#36977] [General]
### Description

When running tasks on Fargate, we need to make sure that capacity provisioning can be done on a per-task basis. Also, it would be nice to have multiple tasks running on one Fargate node, but only one task running on one node.

### Use case/motivation

It's time to go serverless.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by raphaelauv** [comment]
airflow desn't have the feature of kubernetes , airflow do not know and check the requirements of tasks

**Comment by dirrao** [comment]
As of now, we don't have support for server-less executors. As it requires discussion, I am moving it under discussion. We can convert back to the issue once it materializes.

**Reply by o-nikolas** [reply]
https://github.com/apache/airflow/discussions/36977#discussioncomment-8224669

**Comment by potiuk** [comment]
And just to add to it. It's absolutely possible - you could write Fargate executor to do more and figure out what to do. The AWS team works on some AWS-specific executors (cc: @o-nikolas @ferruzzi @vincbeck ) and Fargate executor was on their radar at some point of time, but if someone would like to develop fargate executor with some of the features you mentioned, I think everyone would be happy to have it (it could be you @5pponent for example)

**Reply by o-nikolas** [reply]
We already have this actually! :grinning:

Myself and the folks at Amazon have created the AWS ECS Executor ([docs link](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/executors/ecs-executor.html)) which supports all launch types including Fargate (which is the default actually). The executor is still experimental because we may still make changes to behaviour/API. But you can load it up and start using it today :)

@potiuk @5pponent

**Reply by potiuk** [reply]
Oh. How cool is that - I watched it being developed from afar, but I was not aware Fargate is an option (let alone the default)
